src/Tools/ExternalAccess/Razor/DefaultDocumentServiceProviderFactory.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor
+{
+    [Shared]
+    [Export(typeof(DocumentServiceProviderFactory))]
+    internal class DefaultDocumentServiceProviderFactory : DocumentServiceProviderFactory
```

Comment:
>internal class DefaultDocumentServicePr [](start = 4, length = 39)

Couldn't this be kept in ASP.NET repo? It's good to minimize the amount of code here.

Reply:
Eh, we need pieces of this here. I've put the core guts of this in the ASP.NET Core repo but I had to leave one of these here in order to allow Roslyn to request things like `ISpanMappingService` and then for this to request `IRazorSpanMappingService` from an inner provider and wrap the result in a mapping service wrapper